README.md

```diff
@@ -6,17 +6,98 @@ Explorative sandbox for GPU accelerated C and CUDA, with Boids "bird-oid objects
 
 ## Roadmap
 - Structures for BOID representation:
-   - x, y, dx, dy
+   - x, y, z, dx, dy, dz
 - CUDA Kernel Functions:
+   - Get random initialisation of Positions from the CPU in the beginning.
    - GetNeighbours
    - ParallelSort
    - BOID
       - sensory information: near wall or boundary, close BOIDS with there distance/orientation/velocity
-         - firendly or predetory behaviour? Avoidance or Follow?
+      - firendly or predetory behaviour? Avoidance or Follow?
 - Datastorage for Output:
    - Plain text for now, or binary, nothing fancy.
 - Datadisplay:
-  - External Python
+   - External Python
+ 
+## The program flowchart
+
+```mermaid
+flowchart
+    subgraph GPU with CUDA
+        direction TB
+        GPU_1[Variables]
+        subgraph BOID Logic
+        GPU_6[basic logic function]
+        GPU_2[find_neighbour]
+        GPU_3[hashtable]
+        GPU_4[sort]
+        GPU_5[boundary check]
+        GPU_7[FriendOrFoe check]
+        end
+        GPU_6 --> GPU_2
+        GPU_6 --> GPU_5
+        GPU_6 --> GPU_7
+        GPU_2 --- GPU_3
+        GPU_3 --- GPU_4
+        GPU_6 --read--> GPU_1
+        GPU_6 --update-->GPU_1
+    end
+    subgraph CPU
+        direction TB
+        CPU_1[Initalisation]
+        CPU_2[Variables]
+        CPU_3[save in file]
+        CPU_1 --> CPU_2
+        CPU_2 --after update-->CPU_3
+    end
+CPU_2 --initalise--> GPU_1
+CPU_2 --get update--> GPU_1
+```
+
+## Class and function overview
+<!-- for documentation of mermaid editor https://mermaid.js.org/syntax/classDiagram.html -->
+```mermaid
+classDiagram
+    direction LR
+    class BOID{
+        +float position x, y, z
+        +float velocity dx, dy, dz
+        +int GridID gx, gy, gz
+        +int species
+        +boid_logic()
+        +check_obstacle()
+    }
+
+    note for HashtableLookup "implementation depends if\n functionality is external\n or if BOIDS itself is saved in a gridcell"
+    class HashtableLookup{
+        
+    }
+```
+
+## Aditional Souces
+
+__CUDA - Amazing Lecture notes:__
+
+https://cis565-fall-2021.github.io/syllabus/
+
+https://github.com/CIS565-Fall-2023/Project1-CUDA-Flocking/blob/main/INSTRUCTION.md
```

Comment:
Can you add `[..](https://...` style links to these?

Reply:
i think i did that now.